docs/architecture/adr-076-sign-mode-unified.md

```diff
@@ -0,0 +1,327 @@
+3. The signer verifies the integrity of the received chunks by computing the root hash of the merkle tree from the chunks and comparing it against the tree root hash in the metadata digest.
+4. The signer decodes the transaction fields using the received chunks and displays them to the user.
+5. The user reviews the transaction and signs it.
+6. Once the transaction is broadcasted, the validator compares the metadata hash included in the signed transaction against the one locally stored in the node to verify that all the messages were signed with the consensual metadata structure.
+
+
+
+```mermaid
+sequenceDiagram
+    participant Wallet
+    participant Node
+    participant User
+    participant Validator
+
+    Wallet->>Node: Request root hash
+    Node-->>Wallet: Send root hash
+    Wallet->>Wallet: Build Transaction
+    Wallet->>Node: Request Metadata Chunks
+    Node-->>Wallet: Send Metadata Chunks
+    Wallet->>Wallet: Verify Chunks
+    Wallet->>User: Display Transaction Details
+    User->>Wallet: Approve and Sign
+    Wallet->>Node: Broadcast Signed Transaction
+    Node->>Validator: Forward Transaction
+    Validator->>Validator: Verify Metadata Root Hash
+    Validator->>Validator: Compare with local values
+```
+
+## Decision
+
+We will implement `SIGN_MODE_UNIFIED` as a new sign mode for the Cosmos SDK. This decision involves the following key actions:
+
+* The implementation will include all the necessary code to build and support the new sign mode, including:
+    * Implementation of the MetadataDigest structure and its associated hash calculation function.
+    * Development of the Merkle tree construction and verification logic for the metadata.
+    * Integration of the new sign mode into the existing transaction signing and verification pipeline.
+* A tool will be created to build the metadata file from the existing protobuf files. This tool will:
+    * Parse all relevant protobuf files in the Cosmos SDK.
+    * Generate a structured JSON metadata file that includes all type and message definitions.
+    * A CI job will be implemented to automatically run this metadata generation tool.
+
+* The transaction signing process will be updated to include the MetadataDigest hash in the sign doc.
+* The Ledger HW signing process will be accommodated to match this new sign mode.
+* The transaction verification process will be modified to validate the metadata hash against the locally stored metadata on validator nodes.
+* Comprehensive documentation for `SIGN_MODE_UNIFIED` will be created, including guides for wallet developers and chain developers on how to implement and use the new sign mode.
+* A migration strategy will be implemented to allow for a (progressive) smooth transition from existing sign modes to `SIGN_MODE_UNIFIED`.
+* Finally, all legacy sign modes will be deprecated and removed from the SDK.
+
+## Consequences
+
+### Backwards Compatibility
+
+**Preservation of Existing Sign Modes:**
```

Comment:
i would add that we mark them as deprecated with the idea that in 2026 this will be the only supported type for signing

Reply:
i see you mention it elsewhere